src/main/java/com/stripe/model/EventData.java

```diff
@@ -10,6 +10,22 @@
 @Setter
 @EqualsAndHashCode(callSuper = false)
 public class EventData extends StripeObject {
-  StripeObject object;
+  /**
+   * Raw JSON object intended to be deserialized as {@code StripeObject}. The deserialization
+   * should be deferred to the user. See the now deprecated method {@link EventData#getObject()}.
+   */
+  JsonObject object;
```

Comment:
Having JSON makes sure that class-specific deserialization failure is deferred (not during deserializing the outer class `Event`) and can be handled in the new API `EventVersionedData`.

Although there is type change here, but the exposed API to client getting StripeObject is still the same via `getObject` deprecated method below.